code/game/objects/items/devices/miragedrive.dm

```diff
@@ -0,0 +1,57 @@
+#define COOLDOWN_DASH 20
+#define COOLDOWN_MASH 10
+#define COOLDOWN_HIT 50
+/obj/item/mdrive
+	name = "mirage drive"
+	desc = "A device that functions to increase the user's kinetic energy and direct it. Should the user land near other beings, the device will draw from them, slowing them down \
+	and allowing the device to be used sooner. Although it's too disorienting to be done with high frequency, the user can attack those they land in the immediate vicinity of."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "miragedrive"
+	lefthand_file = 'icons/mob/inhands/misc/devices_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/misc/devices_righthand.dmi'
+	item_state = "mdrive"
+	w_class = WEIGHT_CLASS_SMALL
+	var/next_dash = 0
+	var/next_hit
+	var/mob/living/L
+	var/access_card = new /obj/item/card/id/captains_spare()
```

Comment:
this path name sucks, also the var L doesn't do anything and the access card is ???

Reply:
L is for things the drive affects and the access card is so get_path_to doesn't create runtimes on every turf